tests/IneffectiveJoinDetectorTest.php

```
@@ -0,0 +1,73 @@
+namespace Koriym\SqlQuality;
+
+use Koriym\SqlQuality\Detector\IneffectiveJoinDetector;
+use PHPUnit\Framework\TestCase;
+
+final class IneffectiveJoinDetectorTest extends TestCase
+{
+    public function testDetectsPlainNestedLoopJoinWithoutGroupingOperation(): void
+    {
+        $detector = new IneffectiveJoinDetector();
+
+        $this->assertTrue($detector->detect([
```

PHPStan (2.2.1)

[error] Parameter `#1` $explainResult of method Koriym\SqlQuality\Detector\IneffectiveJoinDetector::detect() expects array{query_block: array{select_id: int, cost_info?: array{query_cost?: float, read_cost?: float, eval_cost?: float, sort_cost?: float, tmp_table_cost?: float}, table?: array{table_name: string, access_type: string, possible_keys?: string|null, key?: string|null, rows: int, rows_examined_per_scan?: int, filtered?: float, attached_condition?: string, ...}, grouping_operation?: array{using_temporary_table?: bool, using_filesort?: bool, cost_info?: array{query_cost?: float, read_cost?: float, eval_cost?: float, sort_cost?: float, tmp_table_cost?: float}, table?: array{table_name: string, access_type: string, possible_keys?: string|null, key?: string|null, rows: int, rows_examined_per_scan?: int, filtered?: float, attached_condition?: string, ...}, nested_loop?: array<array{table: array{table_name: string, access_type: string, possible_keys?: string|null, key?: string|null, rows: int, rows_examined_per_scan?: int, filtered?: float, attached_condition?: string, ...}}>, duplicates_removal?: array{using_temporary_table?: bool, using_filesort?: bool, table?: array{table_name: string, access_type: string, possible_keys?: string|null, key?: string|null, rows: int, rows_examined_per_scan?: int, filtered?: float, attached_condition?: string, ...}}}, ordering_operation?: array{using_temporary_table?: bool, using_filesort?: bool, cost_info?: array{query_cost?: float, read_cost?: float, eval_cost?: float, sort_cost?: float, tmp_table_cost?: float}, table?: array{table_name: string, access_type: string, possible_keys?: string|null, key?: string|null, rows: int, rows_examined_per_scan?: int, filtered?: float, attached_condition?: string, ...}, nested_loop?: array<array{table: array{table_name: string, access_type: string, possible_keys?: string|null, key?: string|null, rows: int, rows_examined_per_scan?: int, filtered?: float, attached_condition?: string, ...}}>, duplicates_removal?: array{using_temporary_table?: bool, using_filesort?: bool, table?: array{table_name: string, access_type: string, possible_keys?: string|null, key?: string|null, rows: int, rows_examined_per_scan?: int, filtered?: float, attached_condition?: string, ...}}}, duplicates_removal?: array{using_temporary_table?: bool, using_filesort?: bool, table?: array{table_name: string, access_type: string, possible_keys?: string|null, key?: string|null, rows: int, rows_examined_per_scan?: int, filtered?: float, attached_condition?: string, ...}}, select_list_subqueries?: array<array{query_block: array{table: array{table_name: string, access_type: string, possible_keys?: string|null, key?: string|null, rows: int, rows_examined_per_scan?: int, filtered?: float, attached_condition?: string, ...}}}>, nested_loop?: array{using_temporary_table?: bool, using_filesort?: bool, cost_info?: array{query_cost?: float, read_cost?: float, eval_cost?: float, sort_cost?: float, tmp_table_cost?: float}, table?: array{table_name: string, access_type: string, possible_keys?: string|null, key?: string|null, rows: int, rows_examined_per_scan?: int, filtered?: float, attached_condition?: string, ...}, nested_loop?: array<array{table: array{table_name: string, access_type: string, possible_keys?: string|null, key?: string|null, rows: int, rows_examined_per_scan?: int, filtered?: float, attached_condition?: string, ...}}>, duplicates_removal?: array{using_temporary_table?: bool, using_filesort?: bool, table?: array{table_name: string, access_type: string, possible_keys?: string|null, key?: string|null, rows: int, rows_examined_per_scan?: int, filtered?: float, attached_condition?: string, ...}}}, ...}, analyze_result: array<string, mixed>}, array{query_block: array{nested_loop: array{array{table: array{table_name: 'posts', access_type: 'ALL', rows_examined_per_scan: 5000, rows_produced_per_join: 100}}, array{table: array{table_name: 'comments', access_type: 'ref', rows_examined_per_scan: 1, rows_produced_per_join: 1}}}}} given.
Offset 'query_block' (array{select_id: int, cost_info?: array{query_cost?: float, read_cost?: float, eval_cost?: float, sort_cost?: float, tmp_table_cost?: float}, table?: array{table_name: string, access_type: string, possible_keys?: string|null, key?: string|null, rows: int, rows_examined_per_scan?: int, filtered?: float, attached_condition?: string, ...}, grouping_operation?: array{using_temporary_table?: bool, using_filesort?: bool, cost_info?: array{query_cost?: float, read_cost?: float, eval_cost?: float, sort_cost?: float, tmp_table_cost?: float}, table?: array{table_name: string, access_type: string, possible_keys?: string|null, key?: string|null, rows: int, rows_examined_per_scan?: int, filtered?: float, attached_condition?: string, ...}, nested_loop?: array<int|string, array{table: array{table_name: string, access_type: string, possible_keys?: string|null, key?: string|null, rows: int, rows_examined_per_scan?: int, filtered?: float, attached_condition?: string, ...}}>, duplicates_removal?: array{using_temporary_table?: bool, using_filesort?: bool, table?: array{table_name: string, access_type: string, possible_keys?: string|null, key?: string|null, rows: int, rows_examined_per_scan?: int, filtered?: float, attached_condition?: string, ...}}}, ordering_operation?: array{using_temporary_table?: bool, using_filesort?: bool, cost_info?: array{query_cost?: float, read_cost?: float, eval_cost?: float, sort_cost?: float, tmp_table_cost?: float}, table?: array{table_name: string, access_type: string, possible_keys?: string|null, key?: string|null, rows: int, rows_examined_per_scan?: int, filtered?: float, attached_condition?: string, ...}, nested_loop?: array<int|string, array{table: array{table_name: string, access_type: string, possible_keys?: string|null, key?: string|null, rows: int, rows_examined_per_scan?: int, filtered?: float, attached_condition?: string, ...}}>, duplicates_removal?: array{using_temporary_table?: bool, using_filesort?: bool, table?: array{table_name: string, access_type: string, possible_keys?: string|null, key?: string|null, rows: int, rows_examined_per_scan?: int, filtered?: float, attached_condition?: string, ...}}}, duplicates_removal?: array{using_temporary_table?: bool, using_filesort?: bool, table?: array{table_name: string, access_type: string, possible_keys?: string|null, key?: string|null, rows: int, rows_examined_per_scan?: int, filtered?: float, attached_condition?: string, ...}}, select_list_subqueries?: array<int|string, array{query_block: array{table: array{table_name: string, access_type: string, possible_keys?: string|null, key?: string|null, rows: int, rows_examined_per_scan?: int, filtered?: float, attached_condition?: string, ...}}}>, nested_loop?: array{using_temporary_table?: bool, using_filesort?: bool, cost_info?: array{query_cost?: float, read_cost?: float, eval_cost?: float, sort_cost?: float, tmp_table_cost?: float}, table?: array{table_name: string, access_type: string, possible_keys?: string|null, key?: string|null, rows: int, rows_examined_per_scan?: int, filtered?: float, attached_condition?: string, ...}, nested_loop?: array<int|string, array{table: array{table_name: string, access_type: string, possible_keys?: string|null, key?: string|null, rows: int, rows_examined_per_scan?: int, filtered?: float, attached_condition?: string, ...}}>, duplicates_removal?: array{using_temporary_table?: bool, using_filesort?: bool, table?: array{table_name: string, access_type: string, possible_keys?: string|null, key?: string|null, rows: int, rows_examined_per_scan?: int, filtered?: float, attached_condition?: string, ...}}}, ...}) does not accept type array{nested_loop: array{array{table: array{table_name: 'posts', access_type: 'ALL', rows_examined_per_scan: 5000, rows_produced_per_join: 100}}, array{table: array{table_name: 'comments', access_type: 'ref', rows_examined_per_scan: 1, rows_produced_per_join: 1}}}}: Array does not have offset 'select_id'.

(argument.type)